selfie.c

```diff
@@ -1508,24 +1508,36 @@ int atoi(int* s) {
       // c was not a decimal digit
       return -1;
 
-    // assert: s contains a decimal number, that is, with base 10
-    n = n * 10 + c;
+    // assert: s contains a decimal number
+
+    // use base 10 but avoid integer overflow
+    if (n < INT_MAX / 10)
+      n = n * 10 + c;
+    else if (n == INT_MAX / 10) {
+      // In decimal representation of integers,
```

Comment:
I see. I wanted to describe, that the least significant digit in 2's kompliment (with base 10) of INT_MIN is indeed exactly 1 greater than the least significant digit of INT_MAX. This is not the case for a 2's kompliment representation with e.g. a base of 16. So we can only rely on this fact for decimal integer literals. 

Anyway, i'll remove the comment.


![dez2erkompl](https://user-images.githubusercontent.com/17096190/28935292-62f11024-7884-11e7-8395-2e341f0e4829.PNG)